src/OrleansProviders/SQLServer/CreateTables.sql

```diff
@@ -1,113 +1,1440 @@
+	);
+
+	CREATE TABLE [OrleansSiloMetricsTable]
+	(
+		[DeploymentId] NVARCHAR(150) NOT NULL, 
+		[SiloId] NVARCHAR(150) NOT NULL, 
+		[Timestamp] DATETIME2(7) NOT NULL, 
+		[Address] VARCHAR(45) NOT NULL, 
+		[Port] INT NOT NULL, 
+		[Generation] INT NOT NULL, 
+		[HostName] NVARCHAR(150) NOT NULL, 
+		[GatewayAddress] VARCHAR(45) NULL, 
+		[GatewayPort] INT NULL, 
+		[CPU] FLOAT NOT NULL,
+		[Memory] BIGINT NOT NULL,
+		[Activations] INT NOT NULL,
+		[RecentlyUsedActivations] INT NOT NULL,
+		[SendQueue] INT NOT NULL, 
+		[ReceiveQueue] INT NOT NULL, 
+		[RequestQueue] BIGINT NOT NULL,
+		[SentMessages] BIGINT NOT NULL,
+		[ReceivedMessages] BIGINT NOT NULL,
+		[LoadShedding] BIT NOT NULL,
+		[ClientCount] BIGINT NOT NULL,
+    
+		CONSTRAINT PK_OrleansSiloMetricsTable_DeploymentId_SiloId PRIMARY KEY([DeploymentId], [SiloId]),
+		CONSTRAINT FK_OrleansSiloMetricsTable_OrleansMembershipVersionTable_DeploymentId FOREIGN KEY([DeploymentId]) REFERENCES [OrleansMembershipVersionTable]([DeploymentId])
+	);
+
+	-- Some of the Orleans queries are version specific due to ROWVERSION in SQL Server 2005 and later.
+	-- ROWVERSION is applied automatically whereas an etag mechanism of using UNIQUEIDENTIFIER in SQL Server is not.
+	-- Also some queries could be tuned better on SQL Server 2005 and later such as error handling or SQL Server 2008
+	-- and later using MERGE for UPSERT (reminders).
+	INSERT INTO [OrleansQuery]([Key], [Query], [Description])
+	VALUES
+	(
+		'UpdateIAmAlivetimeKey',
+		N'
+		-- This is not expected to never fail by Orleans, so return value
+		-- is not needed nor is it checked.
+		SET NOCOUNT ON;
+		BEGIN TRANSACTION;
+		UPDATE [OrleansMembershipTable]
+		SET
+			IAmAliveTime = @iAmAliveTime	    
+		WHERE
+			([DeploymentId] = @deploymentId AND @deploymentId IS NOT NULL)
+			AND ([Address] = @address AND @address IS NOT NULL)
+			AND ([Port] = @port AND @port IS NOT NULL)
+			AND ([Generation] = @generation AND @generation IS NOT NULL);
+		COMMIT TRANSACTION;',
+		N''
+	);
+
+	INSERT INTO [OrleansQuery]([Key], [Query], [Description])
+	VALUES
+	(
+		'InsertMembershipVersionKey',
+		N'SET NOCOUNT ON;
+		BEGIN TRANSACTION;
+		INSERT INTO [OrleansMembershipVersionTable]
+		(
+			[DeploymentId],
+			[Timestamp],
+			[Version]	    
+		)
+		SELECT	
+			@deploymentId,
+			SYSUTCDATETIME(),
+			@version        
+		WHERE NOT EXISTS
+		(
+			SELECT 1
+			FROM OrleansMembershipVersionTable
+			WHERE [DeploymentId] = @deploymentId AND @deploymentId IS NOT NULL
+		);
+                                        
+		IF @@ROWCOUNT > 0
+		BEGIN
+			COMMIT TRANSACTION;
+			SELECT CAST(1 AS BIT);
+		END
+		ELSE
+		BEGIN
+			ROLLBACK TRANSACTION;
+			SELECT CAST(0 AS BIT);
+		END',
+		N''
+	);
+
+	INSERT INTO [OrleansQuery]([Key], [Query], [Description])
+	VALUES
+	(
+		'InsertMembershipKey',
+		N'SET NOCOUNT ON;
+		BEGIN TRANSACTION; --  @@TRANCOUNT = 0 -> +1.
```

Comment:
This way of doing transaction handling is explicit and SQL Server 2000 compatible. It might be more clear to use try-catch in SQL Server 2005 and newer, maybe not. At least this is explicit on what is happening.